src/billing/billing-address-reducer.ts

```diff
@@ -58,13 +58,13 @@ function errorsReducer(
     case BillingAddressActionType.UpdateBillingAddressFailed:
         return objectSet(errors, 'updateError', action.payload);
 
-    case CustomerActionType.UpdateCustomerRequested:
-    case CustomerActionType.UpdateCustomerSucceeded:
+    case SubscriptionsActionType.UpdateSubscriptionsRequested:
```

Comment:
Since update subscriptions is a separate action, I think it'd be better to store the status and error separately than storing them against `continueAsGuestError` and `isContinuingAsGuest` respectively, and have a separate selector each to reflect these values. For example, it'd be odd for the caller to get `isContinuingAsGuest` as `true` when `updateSubscriptions` is called; instead, the caller would expect to call `isUpdatingSubscriptions` to get such value.

Reply:
Also related to your other comment, I think the intent here was to provide an easy way to continue as guest and subscribe that email to newsletter, in a parallel, by using a single SDK action.

Since I was merging these two actions (update billing address & update subscription), I was running into a problem that `isContinuingAsGuest` could get stuck in `true` when updating subscription failed, that's why I'm doing everything here (https://github.com/bigcommerce/checkout-sdk-js/pull/805/files)

Maybe I could still keep a new the state separately, but I think I might need to keep these listeners here?

Reply:
Right I see what you mean, both `continueAsGuest` and `updateSubscriptions` actions update the customer's subscription consent. I think what you should do is to, as you said, keep what you have here, but also create a new reducer and selector to specifically store and retrieve the value for subscription updates. With them, the caller can specifically retrieve the status and error state for the new `updateSubscriptions` method.